Modules/IO/ImageBase/include/itkImageSeriesReader.hxx

```diff
@@ -457,17 +434,20 @@ ImageSeriesReader<TOutputImage>::GenerateData()
     // Deep copy the MetaDataDictionary into the array
     if (reader->GetImageIO() && needToUpdateMetaDataDictionaryArray)
     {
-      auto newDictionary = new DictionaryType;
-      *newDictionary = reader->GetImageIO()->GetMetaDataDictionary();
+      MetaDataDictionary newDictionary = reader->GetImageIO()->GetMetaDataDictionary();
       if (nonUniformSampling)
       {
         // slice-specific information
-        EncapsulateMetaData<double>(*newDictionary, "ITK_non_uniform_sampling_deviation", spacingDeviation);
+        EncapsulateMetaData<double>(newDictionary, "ITK_non_uniform_sampling_deviation", spacingDeviation);
       }
-      m_MetaDataDictionaryArray.push_back(newDictionary);
+      m_InternalMetaDataDictionaries.push_back(newDictionary);
```

Comment:
Copying the dictionary is a non-trivial task. The prior implementation used pointer to ensure extra copying was not done. This now needs to manually be done.

This should be "emplace_back" to reduce copying the dictionary.  Also when the vector is resize, it will copy the dictionaries. The vector should reserve the expected number of dict ( aka number of slices.

Reply:
@blowekamp Thanks for expressing your concern, but I think you are mistaken. Copying a dictionary is actually pretty fast, now that it has "copy on write". For more than six years already: 

- Look at pull request #455 commit 327de92e76d7f228fbca6aaf6c8e679f6638d6ce (especially look at the author 😺)!

No problem, I can add a `vector::reserve` call, and a `std::move`, but I don't expect a major performance change 🤷

Reply:
@blowekamp Please check if your concern is taken away by this [force-push](https://github.com/InsightSoftwareConsortium/ITK/compare/b12865273b8fe1be5094684cdae4995ecc0a46bc..d7150cdf6f9fae17da67a0eedaf07505f2e081b2)

Reply:
Ahh... Thanks for the link, lol. And thank you for using the move semantics which is fine, I agree it will not make much difference given the "brilliant" implement in the DataDictionary 🤣 


Side details discussion: I do find your C++ changes interesting and frequently learn things from them. Thank you. I suppose resizing and vector where the objects support move semantics is not too expensive, even without lazy copying.